spec/unit/tariff_synchronizer/base_update_spec.rb

```diff
@@ -233,4 +233,39 @@
       it { is_expected.to be_nil }
     end
   end
+
+  describe '.by_filename' do
+    subject(:update) {}
+
+    before do
+      allow(TradeTariffBackend).to receive(:service).and_return(service)
+    end
+
+    let(:uk_update) { create(:cds_update) }
+    let(:xi_update) { create(:taric_update) }
+
+    context 'when on the uk service and fetching the xi update' do
+      let(:service) { 'uk' }
+
+      it { expect { described_class.by_filename(xi_update.to_param) }.to raise_error(Sequel::RecordNotFound) }
+    end
+
+    context 'when on the uk service and fetching the uk update' do
+      let(:service) { 'uk' }
+
+      it { expect(described_class.by_filename(uk_update.to_param)).to be_a(TariffSynchronizer::CdsUpdate) }
+    end
```

Comment:
It would be nice to make explicit that when you fetch the UK update, it'll get the GZIP file

Reply:
and when you fetch the XI it'll use the XML.

Reply:
I deliberately pushed the logic down to the model so we don't have to orchestrate all of that

Reply:
Misunderstood the point, here. Will have a look about making it more explicit.

Reply:
So thinking about  it, that should be an implementation detail that the caller does not know about.